packages/flutter/test/cupertino/search_field_test.dart

```diff
@@ -111,7 +111,24 @@ void main() {
       );
     },
   );
-
+  testWidgets(
+    'can change keyboard type',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        CupertinoApp(
+          home: Center(
+            child: CupertinoSearchTextField(
+              keyboardType: TextInputType.number
+            ),
+          ),
+        ),
+      );
+      await tester.tap(find.byType(CupertinoSearchTextField));
+      await tester.showKeyboard(find.byType(CupertinoSearchTextField));
+       expect(tester.testTextInput.setClientArgs['inputType']['name'],
+        equals('TextInputType.number'));
+    },
```

Comment:
Maybe we want to assert that the keyboardType passed into `CupertinoSearchTextField` is passed down to the inner widget (`CupertinoTextField`)?

(just a suggestion)

Reply:
@pedromassango I have no idea how to do that. Can you please help?

Reply:
It would be something like:

1. Find the inner widget (the Cupertinotexy field)
2. Assert that it's `keyboardType` is the same as the one you provided in the Cupertino search text field.